src/Asset.js

```diff
@@ -126,6 +126,11 @@ class Asset {
   async getPackage() {
     if (!this._package) {
       this._package = await this.getConfig(['package.json']);
+      if (this._package) {
+        this._package.pkgdir = path.dirname(
+          await config.resolve(this.name, ['package.json'])
+        );
+      }
```

Comment:
I actually liked your original solution using the resolver. This one will result in two lookups.